dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java

```diff
@@ -214,6 +217,15 @@ default SourceFile getSourceFileByPath(long entryId, String tag, SourceFile.File
             SourceFile.FileType fileType) {
         T entry = getDAO().findById(workflowId);
         checkEntry(entry);
+        // tighten permissions for hosted tools and workflows
```

Comment:
You went out of your way to do this, so it must be right, but I don't understand the code well enough to know why it's OK with non-published entries to get source files, but it is not OK for hosted entries.

Maybe add a comment explaining why here, unless it's obvious to everybody else.

Reply:
@coverbeck 
It needs to go somewhere, I think somewhere higher level though. Will get it on the next pass. 
The story is that getting content (like source files) from workflows up to this point was permitted to be fairly lax. After all, we're only indexing public repos and proxying their content, it doesn't actually matter if you can grab unpublished content from someone else. Its already "public" somewhere else. 

That changes with hosted workflows where unpublished really means that no one else should be able to get their content so we need to be more careful in some situations.